### PR TITLE
ref(ui): Simplify form FieldGroup

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Remove the option of using a render function as the child of a FieldGroup. There is no need to for this as the only time we use this is in the FormField itself where we use it to render the same ControlWrapper (renamed from FieldControl) but with a different controlState component.

The FieldGroup already had types that allowed for the controlState to be overridden in the FieldGroup.

Now the FormField just passes it's custom FormFieldControlState to the FieldGroup's controlState prop instead of rendering the whole ControlWrapper with the custom control statet pr

### CODE REVIEW

File: static/app/components/forms/fieldGroup/index.tsx [hunk 1]
```diff
@@ -58,28 +58,13 @@ function FieldGroup({
     return null;
   }
 
+  const controlStateElement = controlState ?? (
```
Comment (JonasBa) [inline static/app/components/forms/fieldGroup/index.tsx:61]: Drive-by PSA that we should start inlining this. It's annoying to read the render functions of components by having to jump to these definitions (especially as components get larger)

**Review by vuluongj20 — APPROVED** [review]
This is a good update! In general we can be more opinionated and not accommodate unnecessary customizations like the one you're removing here — I don't see why we would ever want to customize `ControlState`.

File: static/app/components/forms/fieldGroup/types.tsx [hunk 2]
```diff
@@ -115,20 +110,5 @@ export interface FieldGroupProps {
 }
 
 /**
- * The children render props mostly pass down FieldGroupProps, with some slight
- * differences for properities that were resolved.
+ * Proops provided to the FieldGroupProps['controlWrapper']
```
Comment (priscilawebdev) [inline static/app/components/forms/fieldGroup/types.tsx:113]:
```suggestion
 * Props provided to the FieldGroupProps['controlWrapper']
```

File: static/app/components/forms/fieldGroup/index.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/components/forms/fieldGroup/index.tsx:61]: I agree with @JonasBa  it would be better if this were inline
